src/Spago/Dhall.hs

```diff
@@ -50,14 +50,15 @@ readRawExpr pathText = do
     else (pure Nothing)
 
 
-writeRawExpr :: Text -> (Text, DhallExpr Dhall.Import) -> IO ()
-writeRawExpr pathText (header, expr) = do
+writeRawExpr :: Bool -> Text -> (Text, DhallExpr Dhall.Import) -> IO ()
+writeRawExpr hasDontFormatConfig pathText (header, expr) = do
   -- After modifying the expression, we have to check if it still typechecks
   -- if it doesn't we don't write to file.
   resolvedExpr <- Dhall.Import.load expr
-  throws (Dhall.TypeCheck.typeOf resolvedExpr)
-  writeTextFile (pathFromText pathText) $ prettyWithHeader header expr <> "\n"
-  format pathText
+  _ <- throws (Dhall.TypeCheck.typeOf resolvedExpr)
+  unless hasDontFormatConfig $ do
```

Comment:
Since we only want to disable the formatting (note that we're skipping also the write here) we should move this condition inside the `format` function

Reply:
I tried this, but `spago build -F` still ends up formatting `spago.dhall`.

Reply:
Is there some other way to avoid this? Should I instead check if we can avoid writing the file as part of a `build`?

Reply:
Oh I see now! The formatting still happens because we automatically migrate old versions of the config to the newest versions [here](https://github.com/spacchetti/spago/blob/75a3407275aa46f73b4f0e5d90049eb5dd42289b/src/Spago/Config.hs#L111): we read in the AST, apply a transformation (`addSourcePaths` in this case) and write the file back. I guess we could check if the config is at the latest version and only migrate if it's not.
Though there are going to be some more changes around that piece of code (with #301) so let's merge this PR and I'll fix this detail while rebasing #301

Reply:
Oh :thinking: alright! Thanks @f-f . I will keep an eye on #301 to know how you did it.